tasties_app/views.py

```diff
@@ -94,5 +94,19 @@ def view_recipe(request, recipe_id=None):
     ingredients = recipe.ingredient_set.all()
     rating = recipe.rating_set.aggregate(Avg('rating'))['rating__avg']
     categories = recipe.categories.all()
-    context = {'recipe': recipe, 'ingredients': ingredients, 'rating': rating, 'categories': categories}
+    if request.method == "POST" and request.POST['action'] == 'Comment':
+        add_comment(request, recipe)
+    comments = recipe.comment_set.all()
+    context = {'recipe': recipe, 'ingredients': ingredients, 'rating': rating,
+               'categories': categories, 'comments': comments}
     return render(request, 'tasties_app/view_recipe.html', context)
+
+
+@login_required(login_url='login')
+def add_comment(request, recipe):
+    comment_value = request.POST.get('comment-adding')
+    user = request.user
+    comment_input = Comment(author_id=user, recipe_id=recipe, publication_date=timezone.now(),
```

Comment:
`comment_input` is a bit of a confusing name in my opinion, I think simply `comment` would have been enough here. Also, specifying `publication_date=timezone.now()` is redundant since it's the default.